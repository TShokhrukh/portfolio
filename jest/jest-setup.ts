import '@testing-library/jest-dom'
import { TextEncoder, TextDecoder } from 'util'
import { IntersectionObserverMock } from './mock/IntersectionObserverMock '

// @ts-ignore
global.TextEncoder = TextEncoder
// @ts-ignore
global.TextDecoder = TextDecoder
// @ts-ignore
global.IntersectionObserver = IntersectionObserverMock
