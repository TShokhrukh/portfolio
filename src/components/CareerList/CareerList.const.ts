import { cn } from '@bem-react/classname'

export const cnCareerList = cn('CareerList')
