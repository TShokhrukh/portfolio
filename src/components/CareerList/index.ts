export * from './build'
export * from './__types__'
export { CareerList as CareerListBase } from './CareerList'
