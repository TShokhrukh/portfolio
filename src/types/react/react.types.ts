import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
