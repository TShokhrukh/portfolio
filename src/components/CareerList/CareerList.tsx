import React, { FC } from 'react'
import { List } from '~/components/List/build'
import { CareerListItem } from './Item'
import { cnCareerList } from './CareerList.const'
import type { ICareerListProps } from './__types__'

export const CareerList: FC<ICareerListProps> = ({
  careers,
  className,
  ...props
}) => {
  return (
    <List {...props} items={careers} itemPropKey='career'
      itemComponent={CareerListItem}
      className={cnCareerList({}, [className])} />
  )
}
