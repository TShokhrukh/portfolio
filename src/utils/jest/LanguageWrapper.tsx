import React, { ReactNode } from 'react'
import { TranslationsProvider } from '@eo-locale/react'

import type { Locale } from '@eo-locale/core'

interface IWrapperProps {
  children: ReactNode,
  language?: string,
  locales?: Locale[]
}

const defLocales: Locale[] = [
  {
    language: 'en',
    messages: {}
  }
]

export const LanguageWrapper: React.FC<IWrapperProps> = ({
  children,
  language = 'en',
  locales = defLocales
}) => {
  return (
    <TranslationsProvider language={language} locales={locales} onError={() => null}>
      {children}
    </TranslationsProvider>
  )
}
