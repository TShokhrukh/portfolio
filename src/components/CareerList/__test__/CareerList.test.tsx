/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { LanguageWrapper } from '~/utils/jest/LanguageWrapper'
import { CareerList } from '../CareerList'
import type { TCareersList } from '../__types__'

describe('CareerList', () => {
  test('must return the component template', () => {
    const careers: TCareersList = [
      {
        id: '1',
        active: true,
        title: 'Title 1',
        desc: {
          en: 'desc 1'
        },
        startDate: '2010-05-09',
        endDate: '2015-06-04'
      },
      {
        id: '2',
        active: false,
        title: 'Title 2',
        desc: {
          en: 'desc 2'
        },
        startDate: '2016-04-09',
        endDate: '2023-06-04'
      }
    ]
    render(
      <LanguageWrapper>
        <CareerList careers={careers} data-testid='career_list' />
      </LanguageWrapper>
    )

    expect(screen.getByTestId('career_list')).toMatchSnapshot()
  })
})
