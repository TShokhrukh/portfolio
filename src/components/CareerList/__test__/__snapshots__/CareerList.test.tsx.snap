// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareerList must return the component template 1`] = `
<div
  class="List CareerList"
  data-testid="career_list"
>
  <div
    class="Card CareerCard CareerCard_active CareerList-Item"
  >
    <h5
      class="Typography Typography_color_white Typography_variant_h5 CareerCard-Title"
    >
      Title 1
    </h5>
    <p
      class="Typography CareerCard-Desc"
    >
      desc 1
    </p>
    <div
      class="CareerCard-Working"
    >
      <p
        class="Typography Typography_color_purple CareerCard-Duration"
      >
        careerCard.durationYears
      </p>
      <p
        class="Typography Typography_color_purple CareerCard-Date"
      >
        2010/May
         - 
        2015/June
      </p>
    </div>
  </div>
  <div
    class="Card CareerCard CareerList-Item"
  >
    <h5
      class="Typography Typography_color_white Typography_variant_h5 CareerCard-Title"
    >
      Title 2
    </h5>
    <p
      class="Typography CareerCard-Desc"
    >
      desc 2
    </p>
    <div
      class="CareerCard-Working"
    >
      <p
        class="Typography Typography_color_purple CareerCard-Duration"
      >
        careerCard.durationYears
        careerCard.durationMonths
      </p>
      <p
        class="Typography Typography_color_purple CareerCard-Date"
      >
        2016/April
         - 
        2023/June
      </p>
    </div>
  </div>
</div>
`;
